File: rb/features.rb
# Sepomex SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SepomexFeatures
  def self.make_feature(name)
    case name
    when "base"
      SepomexBaseFeature.new
    when "test"
      SepomexTestFeature.new
    else
      SepomexBaseFeature.new
    end
  end
end

File: py/features.py
# Sepomex SDK feature factory

from feature.base_feature import SepomexBaseFeature
from feature.test_feature import SepomexTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SepomexBaseFeature(),
        "test": lambda: SepomexTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
